include/mbedtls/pk.h

```diff
@@ -251,6 +282,11 @@ static inline size_t mbedtls_pk_get_len( const mbedtls_pk_context *ctx )
  * \param ctx       Context to test
  * \param type      Target type
  *
+ * \note            The return value of this function only depends on the
+ *                  underlying public key type of the given context and gives
+ *                  the same result as the function \c mbedtls_pk_type_convertible. 
+ *                  For more information, see the documentation of the latter.
+ *
```

Comment:
Yes, `mbedtls_pk_can_do` is functionally equivalent `mbedtls_pk_type_convertible` on the underlying PK types. If you'd find the time to have a look at the proposed PK changes (our internal ticket has more details), I would really appreciate it, because you know the library better than I do and while I think the changes clarify matters, it might of course be that I miss something.